### PR TITLE
ci: deploy rcs onto hub cluster and configure managed cluster during …

### PR DESCRIPTION
**Pull request by dvirgilad**
/unhold

### CODE REVIEW

File: hack/setup-hub-cluster.sh [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    requests:
+      cpu: "100m"
+      memory: "100Mi"
+EOF
+
+
+kubectl patch clustermanagementaddon rcs-score --type merge -p \
+'{"spec":{"installStrategy":{"type":"Placements","placements":[{"name":"ci-test-placement","namespace":"'"ci-test-ns"'","configs":[{"group":"addon.open-cluster-management.io","resource":"addondeploymentconfigs","name":"rcs-score-deploy-config","namespace":"open-cluster-management-hub"}]}]}}}'
+kubectl patch clustermanagementaddon capp-status-addon --type merge -p \
+'{"spec":{"installStrategy":{"type":"Placements","placements":[{"name":"ci-test-placement","namespace":"'"ci-test-ns"'"}]}}}'
```
Comment (mzeevi) [inline hack/setup-hub-cluster.sh:63]: This can be controlled from the rcs helm chart

File: hack/setup-hub-cluster.sh [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+capprepo=https://github.com/dana-team/container-app-operator
+addonsrepo=https://github.com/dana-team/rcs-ocm-addons
+
+cappimage="ghcr.io/dana-team/container-app-operator:${capprelease}"
+rcsaddonsimage="ghcr.io/dana-team/rcs-ocm-addons:${addonsrelease}"
+addons_url=https://github.com/dana-team/rcs-ocm-addons/releases/download/${addonsrelease}/install.yaml
+certmanagerurl=https://github.com/cert-manager/cert-manager/releases/download/v1.13.3/cert-manager.yaml
+
+#Install cert-manager
+kubectl apply -f ${certmanagerurl}
```
Comment (mzeevi) [inline hack/setup-hub-cluster.sh:20]: Install with helm chart

File: Makefile [hunk 3]
```diff
@@ -226,6 +228,15 @@ yq: $(YQ) ## Download yq locally if necessary.
...
 
+.PHONY: setup-ci-hub
+setup-ci-hub: ## setup hub cluster with RCS
+	$(shell pwd)/hack/setup-hub-cluster.sh $(CAPP_RELEASE) $(ADDONS_RELEASE) $(RCS_RELEASE)
+	rm -rf container-app-operator/
+
+.PHONY: cleanup-hub
+cleanup-hub: helm  ## cleanup hub cluster.
+	$(shell pwd)/hack/hub-cleanup.sh
+
```
Comment (mzeevi) [inline Makefile:239]: I'd rather we have the logic in the Makefile instead of in bash scripts in `hack`.

File: .github/workflows/pr.yaml [hunk 4]
```diff
@@ -82,13 +98,37 @@ jobs:
...
+          kubeconfig: ${{ secrets.BACKEND_CI_HUB_CLUSTER_KUBECONFIG }}
+        if: always()
+
       - name: Undeploy backend
         run: make undeploy
         if: always()
 
+      - name: Cleanup hub
+        run: make cleanup-hub
+        if: always()
```
Comment (mzeevi) [inline .github/workflows/pr.yaml:123]: The CI can potentially fail at any stage, so make sure to switch to the correct kube context before the cleanup

File: hack/setup-hub-cluster.sh [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+ --create-namespace oci://ghcr.io/dana-team/helm-charts/rcs-ocm-addons \
+ --version ${addonsrelease}
+
+
+# Install RCS
+helm upgrade --install rcs-deployer \
+  --namespace rcs-deployer-system \
+  --create-namespace \
+  oci://ghcr.io/dana-team/helm-charts/rcs-ocm-deployer \
+  --version ${rcsrelease}
```
Comment (mzeevi) [inline hack/setup-hub-cluster.sh:34]: Maybe manage that with a `Helmfile`?

**Review by mzeevi — CHANGES_REQUESTED** [review]

File: .github/workflows/pr.yaml (hunk 4 above)
Comment (dvirgilad) [reply .github/workflows/pr.yaml:123]: Since they have `if: always()` shouldn't they run?

Comment (mzeevi) [reply .github/workflows/pr.yaml:123]: Nevermind I missed some lines before

File: hack/setup-hub-cluster.sh (hunk 5 above)
Comment (mzeevi) [reply hack/setup-hub-cluster.sh:34]: The cert-manager, rcs-addons and rcs are all installed on the same cluster, no?

**Pull request closed without merging**